src/main/kotlin/eli/inevitable/Inevitable.kt

```diff
@@ -105,17 +106,17 @@ fun textCell(contents: List<String>,
 //     return if(init == null) builder.finish() else builder.init().finish()
 // }
 
-// /**
-//  * Constructs a divider of formatted text.
-//  * @param columnCount The total number of columns
-//  * @param init An optional initialization lambda used to configure the formatting for the divider.
-//  * @return A finalized [DivFormatter]
-//  */
-// fun textDivider(columnCount: Int,
-//                 init: (DivFormatter.Builder.() -> DivFormatter.Builder)? = null): DivFormatter {
-//     val builder = DivFormatter.Builder(columnCount)
-//     return if(init == null) builder.finish() else builder.init().finish()
-// }
+/**
+ * Constructs a divider of formatted text.
+ * @param columnCount The total number of columns
+ * @param init An optional initialization lambda used to configure the formatting for the divider.
+ * @return A finalized [DivFormatter]
+ */
+fun textDivider(columnCount: Int,
+                init: (DivFormatter.Builder.() -> DivFormatter.Builder)? = null): DivFormatter {
```

Comment:
Is init a `Function<Builder,Builder>` or a `Supplier<Builder>`

shouldn’t the usage of init be `init.apply(builder).finish`

Maybe since it’s a consumer of Builder kotlin can implicitly call apply on the function so that might explain why `builder.init().finish` works

Reply:
So, in Kotlin, you can define your suppliers/consumers to exist only within a specific context. This is part of the type-safe builder pattern that allows you to call from within the context of the larger Builder. It's what lets you define specific DSLs:  [Type-Safe Builders](https://kotlinlang.org/docs/reference/type-safe-builders.html)